lib/Frontend/nvqpp/ConvertDecl.cpp

```diff
@@ -632,8 +632,10 @@ bool QuakeBridgeVisitor::VisitVarDecl(clang::VarDecl *x) {
         return true;
 
       // Assign registerName
-      if (auto meas = initVec.getDefiningOp<quake::MeasurementInterface>())
-        meas.setRegisterName(builder.getStringAttr(x->getName()));
+      if (auto descr = initVec.getDefiningOp<quake::DiscriminateOp>())
+        if (auto meas = descr.getMeasurement()
+                            .getDefiningOp<quake::MeasurementInterface>())
+          meas.setRegisterName(builder.getStringAttr(x->getName()));
```

Comment:
Will there be any situations where this won't work? I.e. one example where it might not work is if any discriminate values be written loaded or stored, which case we'd have to traverse through those, too? (I suspect that is pretty unlikely/impossible in this particular file, but the question is really more broad than this one particular location.)